### PR TITLE
PD-1956: Cell isHeader styles + Storybook updates

### PR DESCRIPTION
**Pull request by spark3-mdb**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-1956](https://jira.mongodb.org/browse/PD-1956)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: .changeset/plenty-readers-pump.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/table': patch
+---
+
+Updating isHeader styles in Cell component
```
Comment (TheSonOfThomp) [inline .changeset/plenty-readers-pump.md:5]: We've technically added a prop tp the `Cell` component. This should be a `minor` bump, and we should add more details in the changelog

File: packages/table/src/Cell.tsx [hunk 2]
```diff
@@ -4,15 +4,18 @@ import { css, cx } from '@leafygreen-ui/emotion';
...
 import { useFontSizeContext } from './FontSizeContext';
 import { useDarkModeContext } from './DarkModeContext';
+import { palette } from '@leafygreen-ui/palette';
 
 export const tdInnerDiv = createDataProp('td-inner-div');
 
 interface HeaderCellProps
   extends HTMLElementProps<'th', HTMLTableHeaderCellElement> {
   isHeader: true;
+  isZebra?: boolean;
```
Comment (TheSonOfThomp) [inline packages/table/src/Cell.tsx:14]: 🤔  I'm not sure `isZebra` is the best prop name for this. If you're not familiar with the phrase "zebra striping" it seems a little silly 😆. I'd go with something a little more clear, like `alternateRowHighlightColor` (though maybe not that verbose)

Comment (TheSonOfThomp) [reply packages/table/src/Cell.tsx:14]: Alternatively, we could do what we do with the Select Options, and expose a different API than we use in the child we render in `React.Children.map`

File: packages/table/src/Table.story.jsx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+          <Cell isHeader={withHeaders}>{datum.name}</Cell>
+          <Cell>{datum.age}</Cell>
+          <Cell>{datum.color}</Cell>
+          <Cell>{datum.location}</Cell>
+        </Row>
+      ),
+}
+
+export const CustomLogic = Template.bind({});
+CustomLogic.args = {
```
Comment (TheSonOfThomp) [inline packages/table/src/Table.story.jsx:72]: There's some weird stuff going on in this story 
Screen Shot 2022-04-28 at 5 09 05 PM

File: packages/table/src/Table.story.jsx [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+              {datum.age > 30 && (
+                <Row>
+                  <Cell isHeader={withHeaders}>
+                    expanded name: {datum.name}
+                  </Cell>
+                  <Cell>expanded age: {datum.age}</Cell>
+                  <Cell>expanded color: {datum.color}</Cell>
+                  <Cell>{datum.location}</Cell>
+                </Row>
```
Comment (TheSonOfThomp) [inline packages/table/src/Table.story.jsx:169]: Expanding rows flicker with a white background when opened

**Review by TheSonOfThomp — CHANGES_REQUESTED** [review]
Comments on `props` are minor, but the styling issues should be fixed

File: packages/table/src/Cell.tsx (hunk 2 above)
Comment (spark3-mdb) [reply packages/table/src/Cell.tsx:14]: HAHA yeah you're right. Renaming to `isAltColor`

File: packages/table/src/Row.tsx [hunk 5]
```diff
@@ -50,8 +50,8 @@ const modeStyles = {
       background-color: ${palette.gray.light2};
       color: ${palette.gray.base};
       cursor: not-allowed;
-      box-shadow: 0 -1px 0 inset ${palette.gray.light1},
-        0 1px 0 inset ${palette.gray.light1};
+      border-top: 1px inset ${palette.gray.light1};
+      border-bottom: 1px inset ${palette.gray.light1};
```
Comment (spark3-mdb) [inline packages/table/src/Row.tsx:54]: Not exactly the same, but effectively the same. Using border so I can have the cell's border over this border.

**Review by TheSonOfThomp — APPROVED** [review]
lgtm 🔥

**Pull request merged**